File: dart/SelectionSort.dart
main(){

  List<int> list = [77,42,86,1,88,100,23,85];

  selectionSort(list);

  print("sorted array is:");
  for (var i = 0; i < list.length; i++) {
    print(list[i]);
  }

}

void selectionSort(List<int> list){
  var n = list.length;

  for(int i = 0; i < n-1; i++){
    var minIndex = i;
    for(int j = i+1; j < n; j++){
      if(list[j] < list[minIndex]){
        minIndex = j;
      }

      int temp = list[minIndex];
      list[minIndex] = list[i];
      list[i] = temp;
    }
  }
}

File: Csharp/SelectionSort.cs
PseudoCode:-

procedure selection sort 
   list  : array of items
   n     : size of list

   for i = 1 to n - 1
   /* set current element as minimum*/
      min = i    
  
      /* check the element to be minimum */
      for j = i+1 to n 
         if list[j] < list[min] then
            min = j;
         end if
      end for

      /* swap the minimum element with the current element*/
      if indexMin != i  then
         swap list[min] and list[i]
      end if
   end for
	
end procedure

Code:-

using System; 
class Solution
{  
    static void sort(int []arr) 
    { 
        int n = arr.Length; 
        for (int i = 0; i < n - 1; i++) 
        { 
            int min_idx = i; 
            for (int j = i + 1; j < n; j++) 
                if (arr[j] < arr[min_idx]) 
                    min_idx = j; 
                    
            int temp = arr[min_idx]; 
            arr[min_idx] = arr[i]; 
            arr[i] = temp; 
        } 
    } 

    static void printArray(int []arr) 
    { 
        int n = arr.Length; 
        for (int i=0; i<n; ++i) 
            Console.Write(arr[i]+" "); 
        Console.WriteLine(); 
    } 

    public static void Main() 
    { 
        int []arr = {64,25,12,22,11}; 
        sort(arr); 
        Console.WriteLine("Sorted array"); 
        printArray(arr); 
    } 
} 
